### ISSUE TITLE
Way to Forms 5

### ISSUE THREAD

**Issue by Chartman123** [nextcloud/forms#2041]
This is a collection of tasks needed for preparation/release of Forms v5.

4.3 release:
- [x] #2071
- [x] #2072 
- [x] #2079
- [x] #2278 
- [x] #2333 

See sub-issues for 5.0 related tasks

**Comment by susnux** [comment]
How about unifying our API? Currently there is not really a pattern in the API routes - especially logical nesting.

For example for forms:

What | Current | New
---|---|---
Get forms | GET `/forms` | GET `/forms`
New form | POST `/form` | POST `/forms` :exclamation: 
Get form | GET `/form/:id:` | GET `/forms/:id:`
Clone form | POST `/form/clone/:id:` | POST `/forms/:id:/clone`
Update form | PATCH `/form/update` | PATCH `/forms/:id:`
Transfer ownership | POST `/form/transfer` | - (just use the patch)
Delete form | DELETE `/form/:id:` | DELETE `/forms/:id:`

			
This is just an example, but I think it would make the API more logical

**Comment by Chartman123** [comment]
Yes I think this would definitely make sense... This was always something illogical to me

**Comment by susnux** [comment]
I do not see any use case for the hash other than the legacy link, probably we can also remove that with v5?

**Comment by Chartman123** [comment]
Should we add the newly organized endpoints already in 4.3 so that users of the API can already adapt to the new structure? 

And regarding the hash: If we remove it, our internal links/URLs would only have the id, right? I think we should at least have a look at the APIs so that we don't mix it there (sometimes hash is needed, sometimes the id).

**Comment by susnux** [comment]
> And regarding the hash: If we remove it, our internal links/URLs would only have the id, right? I think we should at least have a look at the APIs so that we don't mix it there (sometimes hash is needed, sometimes the id).

Yes I would just use the form id, like for files. if you use the internal share link it is just the file id.

**Comment by afflux** [comment]
with the `/partial_form/{hash}` endpoint removed, what's the expected way to figure out the form id over the API? Query `/forms` and find the one that matches?

For me, the internal share link does not contain the id, but the hash.

**Comment by Chartman123** [comment]
> Query `/forms` and find the one that matches?

Yes, you can get the ID from `GET /forms[?type=shared]` if you want to access an owned/shared form.

What's your actual use case?
